### PR TITLE
Added doc strings and comments for the controllers package

### PR DESCRIPTION
**Pull request by ElsaChelala**
https://github.com/ContainerSolutions/externalsecret-operator/issues/136

I found this issue in a previous version of the repo and I thought it would be a good way to learn about the project

### CODE REVIEW

**Comment by knelasevero** [comment]
Looks good @ElsaChelala ! You might wanna check you local git setup. Your commits are not being associated with you correct account. 

Check your ssh key pair, if you are using ssh (and that you public key is added to the desired account). Also be sure to use a git config with the same email as the one in your account :smile_cat: 

```
git config --global user.email "YOUR_EMAIL"
```

**Comment by ElsaChelala** [comment]
Hi @knelasevero, Jon helped me fixed the issue, can you make sure everything looks good? 

Thank you!

File: pkg/controllers/externalsecret/externalsecret_controller.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ type Reconciler struct {
 	ControllerClass string
 }
 
+// Reconcile returns a new ExternalSecret object.
```
Comment (knelasevero) [inline pkg/controllers/externalsecret/externalsecret_controller.go:55]: I would change this to something like:
```
// Reconcile implements the main reconciliation loop 
// for watched objects (ExternalSecret, ClusterSecretStore and SecretStore), 
// and updates/creates a Kubernetes secret based on them
```

**Comment by knelasevero** [comment]
Left a comment about one of the docstrings :)

Besides that one the others seem nice!

Thanks Elsa!

File: pkg/controllers/externalsecret/externalsecret_controller.go (hunk 1 above)
Comment (knelasevero) [reply pkg/controllers/externalsecret/externalsecret_controller.go:55]: I am not sure how the linter handles multiline comments for docstrings like this, try to check it out with `make fmt`

**Pull request merged**